### PR TITLE
Teach `jj squash` `--from/--into`, deprecate `jj move`

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * Graph node symbols are now configurable via `ui.graph.default_node` and `ui.graph.elided_node`.
 
+* `jj squash` now accepts `--from` and `--to` (mutually exclusive with `-r`).
```
Comment (yuja) [inline CHANGELOG.md:23]: nit: `s/--to/--into/`

File: cli/src/commands/squash.rs [hunk 2]
```diff
@@ -65,12 +79,24 @@ pub(crate) fn cmd_squash(
...
-    if parents.len() != 1 {
-        return Err(user_error("Cannot squash merge commits"));
+
+    let source;
+    let destination;
+    if args.from.is_some() || args.into.is_some() {
+        source = workspace_command.resolve_single_rev(args.from.as_deref().unwrap_or("@"))?;
+        destination = workspace_command.resolve_single_rev(args.into.as_deref().unwrap_or("@"))?;
+        if source.id() == destination.id() {
+            return Err(user_error("Source and destination cannot be the same."));
```
Comment (yuja) [inline cli/src/commands/squash.rs:89]: nit: period is usually omitted in error message, but the same message in `cmd_move()` has one.

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +20,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  It can thereby be for all uses cases where `jj move` can be used.
```
Comment (joyously) [inline CHANGELOG.md:24]: "uses cases"

File: README.md [hunk 4]
```diff
@@ -276,7 +276,7 @@ Besides the usual rebase command, there's `jj describe` for editing the
 description (commit message) of an arbitrary commit. There's also `jj diffedit`,
 which lets you edit the changes in a commit without checking it out. To split
 a commit into two, use `jj split`. You can even move part of the changes in a
-commit to any other commit using `jj move`.
+commit to any other commit using `jj squash -i --from X --to Y`.
```
Comment (yuja) [inline README.md:279]: nit: `--into`

**Review by yuja — APPROVED** [review]
nice

File: cli/src/commands/move.rs [hunk 5]
```diff
@@ -62,6 +62,14 @@ pub(crate) fn cmd_move(
     command: &CommandHelper,
     args: &MoveArgs,
 ) -> Result<(), CommandError> {
+    writeln!(
+        ui.warning(),
+        "warning: `jj move` is deprecated; use `jj squash` instead, which is equivalent"
+    )?;
+    writeln!(
+        ui.warning(),
+        "warning: `jj move` will be removed in a future version, and this will be a hard error"
```
Comment (joyously) [inline cli/src/commands/move.rs:71]: Will it be a hard error?  What about the possibility that `move` will exist to be like other VCS, where it moves a file in the tree?

File: cli/src/commands/squash.rs (hunk 2 above)
Comment (martinvonz) [reply cli/src/commands/squash.rs:89]: Done.

File: README.md (hunk 4 above)
Comment (martinvonz) [reply README.md:279]: Done.

**Comment by martinvonz** [comment]
I mentioned this PR on Discord and heard only votes for it, so I'll go ahead and merge this.

**Pull request merged**